rust/src/ssh/parser.rs

```diff
@@ -81,7 +98,10 @@ named!(pub ssh_parse_record_header<SshRecordHeader>,
         pkt_len: verify!(be_u32, |&val| val > 1) >>
         padding_len: be_u8 >>
         msg_code: be_u8 >>
-        (SshRecordHeader{pkt_len, padding_len, msg_code})
+        (SshRecordHeader{pkt_len: pkt_len,
+            padding_len: padding_len,
+            msg_code: num::FromPrimitive::from_u8(msg_code).unwrap_or(MessageCode::SshMsgUndefined)})
```

Comment:
I understand this pattern, but we are losing information, that we might want in future versions (ie detect on `ssh.msh_code = 8`)...
Not sure what the best is...

Reply:
I added all possible message code options. There are few ways to store undefined code:
1. Add logging for undefined message code:
```suggestion
            msg_code: num::FromPrimitive::from_u8(msg_code).unwrap_or(MessageCode::SshMsgUndefined)})msg_code: num::FromPrimitive::from_u8(msg_code).unwrap_or_else(|| { 
                SCLogDebug!("SSH undefined message code {}", msg_code);
                MessageCode::SshMsgUndefined
            })})
```

2. Change SshMsgUndefined enum value to be able to store ```u8``` and add separate function for ```to_u8``` conversion:
```
impl MessageCode {
    fn from_u8(value: u8) -> MessageCode {
        match value {
            1 => SshMsgDisconnect,
            2 => SshMsgIgnore,
            3 => SshMsgUnimplemented,
            4 => SshMsgDebug,
            5 => SshMsgServiceRequest,
            6 => SshMsgServiceAccept,
            20 => SshMsgKexinit,
            21 => SshMsgNewKeys,
            30 => SshMsgKexdhInit,
            31 => SshMsgKexdhReply,
            _ => SshMsgUndefined(value),
    }
}
```

3. Remove enum from ```struct SshRecordHeader``` and use const values.

Which option can you suggest?

Reply:
The second option looks nice (but I am not a Rust expert)

Reply:
Used second option in https://github.com/OISF/suricata/pull/5048